src/app/decorators/MetadataStorage.ts

```diff
@@ -0,0 +1,55 @@
+
+export class MetadataStorageAndValidation {
+ public fieldName: string = '';
+ public classNametoCaptcha = {};
+
+    addValidationMetadata(className:string, metadata: string) {
+        this.classNametoCaptcha[className] = metadata;
+    }
+
+    private async checkRecaptcha(value) {
+        if (!value || !value[this.fieldName]) {
+            return new Promise((resolve, reject) => {
+                resolve({});
+            });
+        } else {
+            const Recaptcha = require('recaptcha-verify');
+            const recaptcha = new Recaptcha({
+                secret: process.env.CAPTCHA_SECRET,
+                verbose: true
+            });
+            return new Promise((resolve, reject) => {
```

Comment:
This is perfect, I think we can move this into a custom validation class and this is the implementation of the validate function.

Reply:
Not sure where one of my comments went, I'm doing something wrong apparently